feature/gribi/ate_tests/base_hierarchical_nhg_update/base_hierarchical_nhg_update_test.go

```diff
@@ -537,18 +580,264 @@ func validateTrafficFlows(t *testing.T, ate *ondatra.ATEDevice, good, bad, lb []
+
+	p2flow := createFlow("Flow Port 1 to Port 2", args.ate, args.top, true, &atePort2)
+	p3flow := createFlow("Flow Port 1 to Port 3", args.ate, args.top, true, &atePort3)
+	p4flow := createFlow("Flow Port 1 to Port 4", args.ate, args.top, true, &atePort4)
+
+	t.Log("Validate primary path traffic recieved at ate port2, ate port3 and no traffic on ate port4")
+	validateTrafficFlows(t, args.ate, nil, []*ondatra.Flow{p4flow}, []*ondatra.Flow{p2flow, p3flow}, 1, args.client, false)
+
+	t.Logf("Adding NH %d for trunk4 via gribi", nh3ID)
+	if deviations.GRIBIMACOverrideWithStaticARP(args.dut) {
+		args.client.AddNH(t, nh3ID, "MACwithInterface", deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: btrunk4, Mac: port4mac, Dest: atePort4.IPv4})
+	} else {
+		args.client.AddNH(t, nh3ID, "MACwithInterface", deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: btrunk4, Mac: port4mac})
+	}
+
+	t.Log("Validate traffic switching from  ate port2, ate port3 to ate port4")
+	validateTrafficFlows(t, args.ate, []*ondatra.Flow{p4flow}, []*ondatra.Flow{p2flow, p3flow}, nil, 2, args.client, true)
+
+	t.Log("Validate traffic switching from  ate port4 back to ate port2 and ate port3")
+	validateTrafficFlows(t, args.ate, nil, []*ondatra.Flow{p4flow}, []*ondatra.Flow{p2flow, p3flow}, 3, args.client, true)
+
+}
+
+// deleteinterfaceconfig unconfigs interfaces.
+func deleteinterfaceconfig(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
+	p2 := dut.Port(t, "port2")
+	p3 := dut.Port(t, "port3")
+	p4 := dut.Port(t, "port4")
+
+	gnmi.Delete(t, dut, d.Interface(p2.Name()).Config())
+	gnmi.Delete(t, dut, d.Interface(p3.Name()).Config())
+	gnmi.Delete(t, dut, d.Interface(p4.Name()).Config())
+}
+
+// configDUTDrain configures ports for drain test.
+func configDUTDrain(t *testing.T, dut *ondatra.DUTDevice) {
+	d := gnmi.OC()
+	p2 := dut.Port(t, "port2")
+	p3 := dut.Port(t, "port3")
+	p4 := dut.Port(t, "port4")
+	btrunk2 = netutil.NextAggregateInterface(t, dut)
+
+	i2 := &oc.Interface{Name: ygot.String(btrunk2)}
+	gnmi.Replace(t, dut, d.Interface(*i2.Name).Config(), configInterfaceDUT(i2, &dutPort2))
+	T2 := configureBundle(t, p2.Name(), *i2.Name)
+	gnmi.Replace(t, dut, gnmi.OC().Interface(p2.Name()).Config(), T2)
+
+	btrunk3 := netutil.NextAggregateInterface(t, dut)
+	i3 := &oc.Interface{Name: ygot.String(btrunk3)}
+	gnmi.Replace(t, dut, d.Interface(*i3.Name).Config(), configInterfaceDUT(i3, &dutPort3))
+	T3 := configureBundle(t, p3.Name(), *i3.Name)
+	gnmi.Replace(t, dut, gnmi.OC().Interface(p3.Name()).Config(), T3)
+
+	btrunk4 := netutil.NextAggregateInterface(t, dut)
+	i4 := &oc.Interface{Name: ygot.String(btrunk4)}
+	gnmi.Replace(t, dut, d.Interface(*i4.Name).Config(), configInterfaceDUT(i4, &dutPort4))
+	T4 := configureBundle(t, p4.Name(), *i4.Name)
+	gnmi.Replace(t, dut, gnmi.OC().Interface(p4.Name()).Config(), T4)
+
```

Comment:
Add code to attach trunk(lag) interface to default network-instance 
`+   if deviations.ExplicitInterfaceInDefaultVRF(dut) {
+       fptest.AssignToNetworkInstance(t, dut, *i2.Name, deviations.DefaultNetworkInstance(dut), 0)
+       fptest.AssignToNetworkInstance(t, dut, *i3.Name, deviations.DefaultNetworkInstance(dut), 0)
+       fptest.AssignToNetworkInstance(t, dut, *i4.Name, deviations.DefaultNetworkInstance(dut), 0)
+   }
+   `

Reply:
Done